File: src/riscv/mappedregs.py
# Copyright 2026 Flavien Solt, Quentin Bordier ETH Zurich.
# Licensed under the General Public License, Version 3.0, see LICENSE for details.
# SPDX-License-Identifier: GPL-3.0-only


MAX_DEVICES = 1023
MAX_HARTS = 15872
PRIORITY_BASE = 0x0
PENDING_BASE = 0x1000
ENABLE_BASE = 0x2000
HART_BASE = 0x200000

# 1 << 25
MSIP_WIDTH = 0x4
MTIMER_OFFSET = 0x4000
SPIKE_CLINT_BASE = 0x2000000

File: rvgen/src/riscv/mappedregs.rs
// Copyright 2026 Flavien Solt, Quentin Bordier ETH Zurich.
// SPDX-License-Identifier: GPL-3.0-only

#![allow(non_camel_case_types)]

pub const MAX_DEVICES: usize = 1023;
pub const MAX_HARTS: usize = 15872;
pub const PRIORITY_BASE: usize = 0;
pub const PENDING_BASE: usize = 4096;
pub const ENABLE_BASE: usize = 8192;
pub const HART_BASE: usize = 2097152;
pub const MSIP_WIDTH: usize = 4;
pub const MTIMER_OFFSET: usize = 16384;
pub const SPIKE_CLINT_BASE: usize = 33554432;
